src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Apprenticeship.cs

```diff
@@ -36,12 +39,43 @@ public ApprenticeshipStatus Status
             }
         }
 
-        
+        public DateTime? CompletionDate { get; set; }
 
         public Apprenticeship()
         {
             DataLockStatus = new List<DataLockStatus>();
             PriceHistory = new List<PriceHistory>();
         }
+
+        public virtual void Complete(DateTime completionDate)
+        {
+            if (Status != ApprenticeshipStatus.Live)
+            {
+                throw new InvalidOperationException("Apprenticeship has to be live to be completed");
```

Comment:
Apprenticeship cannot be _already completed_.

Another invariant could be that it can't _Complete_ before it starts, but this may have little value.

Reply:
Is it the message which is confusing? I think the logic looks okay

Reply:
No it's fine, as above.